turbopack/crates/turbopack-core/src/chunk/available_modules.rs

```diff
@@ -98,4 +108,31 @@ impl AvailableModules {
         }
         Ok(Vc::cell(None))
     }
+
+    #[turbo_tasks::function]
+    pub async fn into_snapshot(&self) -> Result<Vc<AvailableModulesSnapshot>> {
+        let mut combined = FxIndexMap::default();
+        if let Some(parent) = self.parent {
+            for (module, info) in &parent.into_snapshot().await?.modules {
+                combined.insert(*module, *info);
+            }
+        }
+
+        Ok(AvailableModulesSnapshot { modules: combined }.cell())
+    }
+}
+
+#[turbo_tasks::value(serialization = "none")]
+#[derive(Debug, Clone)]
+pub struct AvailableModulesSnapshot {
+    modules: FxIndexMap<ResolvedVc<Box<dyn ChunkableModule>>, AvailableModulesInfo>,
+}
```

Comment:
I would have gone for:

``` rs
struct AvailableModulesSnapshot {
    parent: Option<ReadRef<AvailableModulesSnapshot>>,
    modules: ReadRef<AvailableModuleInfoMap>,
}
```

which avoids cloning and merging the maps, but we can change that later...

Reply:
update it